semgrep-core/tests/OTHER/rules/metavar_comparison_constness.yaml

```diff
@@ -0,0 +1,12 @@
+rules:
+  - id: c-comparing-const-vars
+    languages:
+      - c
+    message: Trying to compare constant variables
+    patterns:
+      - pattern: |
+          $X < $Y
+      - metavariable-comparison:
+          comparison: $X < $Y
+          metavariable: $X
```

Comment:
this makes the metavariable field a bit useless now, if you can use multiple metavars in comparison.

Reply:
We already allowed referencing multiple variables. And indeed I think the only use of `metavariable` here is when we have `strip: true` (the default is `false`) in which case we rewrite `$X` as `int($X)`... and this is probably something the user should write manually. If we get rid of `strip` then we can get rid of `metavariable`.

Reply:
Without having to break any rules, we could make `metavariable` optional and require it only when `strip: true`.

Reply:
+1 to having the user write `int($X)` manually. It would be extremely frustrating to want to do `int($X) < int($Y)` and not